contracts/governance/Vesting/TeamVesting.sol

```diff
@@ -1,29 +1,51 @@
 pragma solidity ^0.5.17;
 pragma experimental ABIEncoderV2;
 
-import "./Vesting.sol";
+import "../../openzeppelin/Ownable.sol";
+import "../../interfaces/IERC20.sol";
+import "../Staking/Staking.sol";
+import "../IFeeSharingProxy.sol";
+import "./IVesting.sol";
+import "../ApprovalReceiver.sol";
+import "./VestingStorage.sol";
+import "../../proxy/Proxy.sol";
 
 /**
  * A regular vesting contract, but the owner (governance) is able to withdraw earlier without a slashing
+ * @dev Vesting contracts shouldn't be upgradable, use Proxy instead of UpgradableProxy
  **/
-contract TeamVesting is Vesting {
+contract TeamVesting is VestingStorage, Proxy {
+	/**
+	 * @notice setup the vesting schedule
+	 * @param _logic the address of logic contract
+	 * @param _SOV the SOV token address
+	 * @param _tokenOwner the owner of the tokens
+	 * @param _cliff the cliff in seconds
+	 * @param _duration the total duration in seconds
+	 * */
 	constructor(
+		address _logic,
 		address _SOV,
 		address _stakingAddress,
 		address _tokenOwner,
 		uint256 _cliff,
 		uint256 _duration,
 		address _feeSharingProxy
-	) public Vesting(_SOV, _stakingAddress, _tokenOwner, _cliff, _duration, _feeSharingProxy) {}
-
-	/**
-	 * @notice withdraws all tokens from the staking contract and forwards them to an address specified by the token owner
-	 * @param receiver the receiving address
-	 * @dev can be called only by owner
-	 * */
-	function governanceWithdrawTokens(address receiver) public {
-		require(msg.sender == address(staking), "unauthorized");
+	) public {
+		require(_SOV != address(0), "SOV address invalid");
+		require(_stakingAddress != address(0), "staking address invalid");
+		require(_tokenOwner != address(0), "token owner address invalid");
+		require(_duration >= _cliff, "duration must be bigger than or equal to the cliff");
+		require(_feeSharingProxy != address(0), "feeSharingProxy address invalid");
 
-		_withdrawTokens(receiver, true);
+		_setImplementation(_logic);
+		_setProxyOwner(address(this));
```

Comment:
It's a nice touch, though it mostly burns gas (proxy Owner doesn't have any functions anyway)

Reply:
agree, added it in a rush :-)
removed